pimd/pim_bsm.c

```diff
@@ -534,33 +533,37 @@ static void pim_instate_pend_list(struct bsgrp_node *bsgrp_node)
 	pim_bsm_rpinfos_free(bsgrp_node->partial_bsrp_list);
 }
 
-static bool is_preferred_bsr(struct pim_instance *pim, struct in_addr bsr,
+static bool is_preferred_bsr(struct pim_instance *pim, pim_addr bsr,
 			     uint32_t bsr_prio)
 {
-	if (bsr.s_addr == pim->global_scope.current_bsr.s_addr)
+	if (!pim_addr_cmp(bsr, pim->global_scope.current_bsr))
 		return true;
 
 	if (bsr_prio > pim->global_scope.current_bsr_prio)
 		return true;
 
 	else if (bsr_prio == pim->global_scope.current_bsr_prio) {
+#if PIM_IPV == 4 || !defined(PIM_V6_TEMP_BREAK)
```

Comment:
Please use `pim_addr_cmp` here - the return value has the same meaning as `memcmp`, and we don't need the `#if`

(And the compare with `ntohl` has the same meaning too)

Reply:
Fixed